### DISCUSSION TITLE
How to expand variable in place?

### DISCUSSION THREAD

**Discussion by DoumanAsh** [nushell/nushell#11926] [Q&A] [answered]
Given following:
```
mut git_prompt = (do { git rev-parse --abbrev-ref HEAD } | complete | $in.stdout | str trim)
git rev-list head...origin/$git_prompt  --left-right --ignore-submodules --count
```

Nushell doesn't expand $git_prompt inside command for some reason
This should be something very simple to do so how to do it?

**Comment by ghost** [comment, marked as answer]
string substitution isn't as short as in posix:
```nu
git rev-list $"head...origin/($git_prompt)"  --left-right --ignore-submodules --count
```
in strings with a dollar in front (`$""`) things inside brackets is interpreted as code and then inserted into the string. (and you will have to escape opening brackets with a backslash if you don't want them to contain commands)

**Reply by DoumanAsh** [reply]
I see thanks
This is bad UX, but cannot be helped I guess
